src/pages/ExamsPage/components/ResetExamAttemptModal.jsx

```diff
@@ -1,22 +1,34 @@
 import PropTypes from 'prop-types';
 
+import { useState } from 'react';
 import {
-  Button, useToggle, ModalDialog, ActionRow,
+  Button, useToggle, ModalDialog, ActionRow, StatefulButton,
 } from '@edx/paragon';
 import { useIntl } from '@edx/frontend-platform/i18n';
 import { useDeleteExamAttempt } from '../hooks';
 import messages from '../messages';
 
-const ResetExamAttemptButton = ({ username, examName, attemptId }) => {
+const ResetButtonProps = {
+  labels: {
+    default: 'Reset',
+    pending: 'Resetting...',
+    complete: 'Reset',
+    error: 'Error',
+  },
+  variant: 'primary',
+};
+
+const ResetExamAttemptModal = ({ username, examName, attemptId }) => {
   const [isOpen, open, close] = useToggle(false);
   const resetExamAttempt = useDeleteExamAttempt();
   const { formatMessage } = useIntl();
+  const [resetButtonStatus, setResetButtonStatus] = useState('');
```

Comment:
Would a reasonable default state be `default` to map to the `Reset` state? Or did you choose not to do that because the default value for the `state` prop for the `StatefulButton` is already `default`?

Reply:
The instance of the button is deleted when the modal is closed, meaning it gets set to the default value every time you open the modal.

Reply:
Ah I see what you're saying now. If the default state for `resetButtonStatus` were pending, then the StatefulButton state would be set to "pending" when the modal opens, then the button would appear in the "pending" state with the loading circle, which wouldn't make sense. See the code below + image for an example.

``` const [rejectButtonStatus, setRejectButtonStatus] = useState('pending');```

![image](https://github.com/edx/frontend-app-exams-dashboard/assets/124631592/f6581610-f4ec-4f2d-8e7d-b9b1940427c9)

The way the StatefulButton works (see: https://paragon-openedx.netlify.app/components/statefulbutton/#basic-usage) is that any states other than `pending`, `complete`, or `error` put the StatefulButton in it's "defualt" state, i.e. where it's just a plain, clickable button. So having the default value for `rejectButtonStatus` as a blank string works fine.

Reply:
Ack, sorry, I made a mistake 🤦. I meant `default` and not `pending`, but your explanation still applies.

I was basically wondering whether we want to be explicit about what the default state of `resetButtonStatus` should be. I lean towards being explicit and setting  `const [resetButtonStatus, setResetButtonStatus] = useState('default');` instead of `const [resetButtonStatus, setResetButtonStatus] = useState('');`, but you're right that they're functionally equivalent.